lib/api/api.js

```diff
@@ -184,14 +186,103 @@ const api = {
             }
             return { returnTagCount, isImplicitDeny };
         }
+        let formDataParser;
+        let fileEventData = null;
+
+        if (apiMethod === 'objectPost') {
+            formDataParser = busboy({ headers: request.headers });
```

Comment:
If you pass wrong headers like, `Content-Type: text/plain` crashes with:
```
Error: Unsupported Content-Type.
    at Busboy.getParserByHeaders (/home/mickael/scality/cloudserver/node_modules/@fastify/busboy/lib/main.js:74:9)
    at new Busboy (/home/mickael/scality/cloudserver/node_modules/@fastify/busboy/lib/main.js:36:23)
    at Busboy (/home/mickael/scality/cloudserver/node_modules/@fastify/busboy/lib/main.js:12:43)
    at Object.callApiMethod (/home/mickael/scality/cloudserver/lib/api/api.js:193:30)
    at routePOST (/home/mickael/scality/cloudserver/node_modules/arsenal/build/lib/s3routes/routes/routePOST.js:62:20)
    at routes (/home/mickael/scality/cloudserver/node_modules/arsenal/build/lib/s3routes/routes.js:225:12)
    at Server.routeRequest (/home/mickael/scality/cloudserver/lib/server.js:143:9)
    at Server.emit (node:events:513:28)
    at Server.emit (node:domain:489:12)
    at parserOnIncoming (node:_http_server:998:12)
```

> [!CAUTION]
> The constructor can throw errors:
> [...]
> Unsupported Content-Type. - The Content-Type isn't one Busboy can parse.
> 
> Missing Content-Type-header. - The provided headers don't include Content-Type at all.
> 
> :arrow_right: https://www.npmjs.com/package/@fastify/busboy

Reply:
If you change `Content-Length` to have it differ from the body size, there is a weird behavior where it logs the full body buffer :bomb: 

``` 
{"name":"S3","time":1720514445976,"error":{"bytesParsed":34657,"code":"HPE_INVALID_METHOD","reason":"Invalid method encountered","rawPacket":{"type":"Buffer","data":[128,94...........,209]}},"level":"info","message":"request rejected","hostname":"mickael-xps","pid":21483}
```

But still execute some code

Reply:
The case where content-length is too small is handled and tested here
https://github.com/scality/cloudserver/pull/5601/files#diff-406efa736d8942ad5f5ba3c3404803f3748ef0d0cd44a20b48f5ce074c3105cdR770

In my experience with AWS, Content-Length is not a form field that is accepted or read on anyway, which is consistent with the doc.

The Content-Length header of the request seems to cause no issue with AWS but I notice I didn't handle this one yet in CS and I'm currently strruggling to catch the error... will update

Reply:
This error log is potentially not related to the POST.
And for the content-length bigger it's most likely related to https://github.com/scality/cloudserver/pull/5601#discussion_r1681565314

Reply:
Is there any further action to take on this conversation thread? For the 'content-length bigger' case, we can continue in the other thread.